Samples/AudioPreprocessing/WinUI/AudioPreprocessing/AudioPreprocessing/Model/PreprocessingModel.cs

```diff
@@ -0,0 +1,136 @@
+﻿using Microsoft.AI.MachineLearning;
+using Microsoft.AI.MachineLearning.Experimental;
+using NAudio.Wave;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Windows.Graphics.Imaging;
+using Windows.Media;
+using Operator = Microsoft.AI.MachineLearning.Experimental.LearningModelOperator;
+
+namespace AudioPreprocessing.Model
+{
+    public class PreprocessModel
+    {
+        const string MicrosoftExperimentalDomain = "com.microsoft.experimental";
+
+        public string AudioPath { get; set; }
+
+        public string MelSpecImagePath { get; set; }
+
+        public SoftwareBitmap GenerateMelSpectrogram(string audioPath)
+        {
+            var signalEnumerable = GetSignalFromFile(audioPath);
+            IList<float> signal = signalEnumerable.Cast<float>().ToList();
+            var rawSoftwareBitmap = GetMelspectrogramFromSignal(signal);
+            return rawSoftwareBitmap;
+        }
+
+        private IEnumerable<float> GetSignalFromFile(string filename)
+        {
+            if (!filename.EndsWith(".wav"))
+            {
+                throw new ArgumentException( String.Format("{0} is not a valid .wav file."));
+            }
+            using (var reader = new AudioFileReader(filename))
+            {
+                var nSamples = reader.Length / sizeof(float);
+                var signal = Array.CreateInstance(typeof(float), nSamples);
+                var read = reader.Read(signal as float[], 0, signal.Length);
+                return (IEnumerable<float>)signal;
+            }
+        }
+
+        static SoftwareBitmap GetMelspectrogramFromSignal(
+            IEnumerable<float> rawSignal,
+            int batchSize = 1,
+            int windowSize = 256,
+            int dftSize = 256,
+            int hopSize = 3,
+            int nMelBins = 1024,
+            int samplingRate = 8192,
+            int amplitude = 5000
+            )
+        {
+            float[] signal = rawSignal.Cast<float>().ToArray();
+
+            //Scale the signal by a given amplitude 
+            for (int i = 0; i < signal.Length; i++) signal[i] = signal[i] * amplitude;
+
+            int signalSize = signal.Length;
+            var nDFT = 1 + (signalSize - dftSize) / hopSize;
+            var onesidedDftSize = (dftSize >> 1) + 1;
+
+            long[] signalShape = { batchSize, signalSize };
+            long[] melSpectrogramShape = { batchSize, 1, nDFT, nMelBins };
+
+            var builder = LearningModelBuilder.Create(13)
+                .Inputs.Add(LearningModelBuilder.CreateTensorFeatureDescriptor("Input.TimeSignal", TensorKind.Float, signalShape))
+                .Outputs.Add(LearningModelBuilder.CreateTensorFeatureDescriptor("Output.MelSpectrogram", TensorKind.Float, melSpectrogramShape));
+            builder
```

Comment:
builder is superfluous here = also it looks like Operators. and Inputs/Output. are on different indents.

Reply:
Corrected